cli/http.go

```diff
@@ -55,49 +56,67 @@ func NewAPIError(resp *http.Response) APIError {
 
 // getAPIClients loads and returns user settings along with two APIClients: one configured for the API
 // server endpoints, and another for the REST api.
-func getAPIClients(o *Options) (*APIClient, *APIClient, *model.CLISettings, error) {
+func getAPIClients(ctx context.Context, o *Options) (*APIClient, *APIClient, *model.CLISettings, error) {
 	settings, err := LoadSettings(o)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
-	// create a client for the API servers
-	APIV2Root := settings.APIServerHost + "/rest/v2"
+	// create client for the REST APIs
+	apiURL, err := url.Parse(settings.APIServerHost)
+	if err != nil {
+		return nil, nil, nil, errors.Errorf("Settings file contains an invalid url: %v", err)
+	}
+
 	ac := &APIClient{
 		APIRoot:   settings.APIServerHost,
-		APIRootV2: APIV2Root,
+		APIRootV2: apiURL.Scheme + "://" + apiURL.Host + "/rest/v2",
 		User:      settings.User,
 		APIKey:    settings.APIKey,
 		UIRoot:    settings.UIServerHost,
 	}
 
-	// create client for the REST APIs
-	apiUrl, err := url.Parse(settings.APIServerHost)
-	if err != nil {
-		return nil, nil, nil, errors.Errorf("Settings file contains an invalid url: %v", err)
-	}
-
 	rc := &APIClient{
-		APIRoot:   apiUrl.Scheme + "://" + apiUrl.Host + "/rest/v1",
-		APIRootV2: apiUrl.Scheme + "://" + apiUrl.Host + "/rest/v2",
+		APIRoot:   apiURL.Scheme + "://" + apiURL.Host + "/rest/v1",
+		APIRootV2: apiURL.Scheme + "://" + apiURL.Host + "/rest/v2",
 		User:      settings.User,
 		APIKey:    settings.APIKey,
 		UIRoot:    settings.UIServerHost,
 	}
 
+	// this operation is just to get the side effect of logging
+	// the degraded mode banner.
+	_, err = getRestClientCommunicator(ctx, settings.APIServerHost)
+	grip.Warning(err)
+
 	return ac, rc, settings, nil
 }
 
 // returns a client and settings for the REST V2 APIs
-func getAPIV2Client(o *Options) (client.Communicator, *model.CLISettings, error) {
+func getAPIV2Client(ctx context.Context, o *Options) (client.Communicator, *model.CLISettings, error) {
 	settings, err := LoadSettings(o)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, errors.WithStack(err)
 	}
 
-	c := client.NewCommunicator(settings.APIServerHost)
+	comm, err := getRestClientCommunicator(ctx, settings.APIServerHost)
+	if err != nil {
+		return nil, nil, errors.WithStack(err)
+	}
+
+	return comm, settings, nil
+}
+
+func getRestClientCommunicator(ctx context.Context, url string) (client.Communicator, error) {
+	c := client.NewCommunicator(url)
+
+	banner, err := c.GetBannerMessage(ctx)
```

Comment:
This won't work for most users because the admin route requires superuser privileges. We either need to create a route/client command that returns just the banner, or change the security requirement to only require a logged in user (I'm not too concerned about other people seeing that specific services are degraded, though you could argue that the route needs to match the UI)

Reply:
I went for allowing auth'd read only access to this route.

Reply:
I think you forgot to push

Reply:
fixed